docs/system-admin-guide/authentication/openid-providers/README.md

```diff
@@ -67,118 +57,230 @@ You can configure the following options.
+
+
+
+#### Step 2: Discovery endpoint
+
+- In the next section, you have the option to specify a discovery endpoint URL to pre-fill some public attributes
+  - For Keycloak, this URL is based on the configured realm name `http://keycloak.example.com:443/realms/{realm}/.well-known/openid-configuration`
+- If you have a discovery endpoint URL, choose **I have a discovery endpoint URL** and fill it in
+- Click **Continue**. With a discovery endpoint URL, OpenProject will try to fetch this information and take you to the next step. Observe the page for error responses in case it cannot connect to the endpoint or the returned information is invalid.
+
+![Discovery endpoint URL](./custom-provider-metadata-discovery.png)
+
+#### Step 3: Advanced configuration
+
+- Unless the metadata endpoint provided these values, you will have to fill out some required endpoint URLs, such as **Authorization endpoint**, **User information endpoint**, and **Token endpoint**.
+- Fill out the **Issuer** field which depends on the provider. For Keycloak, this value would be the realm URL: `http://keycloak.example.com:443/realms/{realm}`
+- Optionally fill out:
+  -  **End session endpoint**, an URL where OpenProject should redirect to terminate a user's session.
+  -  **JWKS URI**. This is the URL of the provider's  JSON Web Key Set document containing e.g., signing keys and certificates.
+  - A custom icon by using a publicly available URL to fetch the logo from.
+- Click **Continue** to validate this form and move to the next step. If there are any errors in this form, they will turn red and inform you about what you need to change.
+
+![Custom OpenID provider advanced configuration in OpenProject](custom-provider-advanced-config.png)
+
+#### Step 5: Client details
+
+In the next section, fill out the client credentials provided from your OpenID Connect provider:
+
+- Fill out **Client ID** and **Client secret**
+- If you want users to be redirected to a separate endpoint _after logging out_ at the identity provider, set **Post Logout Redirect URI**.
+- If you want this login mechanism to respect the global setting for self registration limits, check **Limit self registration**.
+- Click **Continue**.
+
+#### Step 6: Optional attribute mapping
+
+You can optionally provide a custom mapping for attributes in the `userinfo` endpoint response. In most cases, you can leave this empty, unless you are providing custom attributes for user properties.
+
+If you need to set some of these values, enter the attribute key used/returned in the `userinfo` endpoint.
+
+For example: Keycloak allows you to map custom properties of the user. This allows you to specify a login with, e.g, `preferred_username` userinfo. In this case, you would fill out `Mapping for: Username` with TODO
```

Comment:
@oliverguenther I think this sentence was not finished. Fill out with... ? I've added a TODO for easier identification later.